src/common/pty.rs

```diff
@@ -77,16 +77,90 @@ pub enum PtyInstruction {
     Quit,
 }
 
-pub struct PtyBus {
-    pub send_screen_instructions: SenderWithContext<ScreenInstruction>,
-    pub send_plugin_instructions: SenderWithContext<PluginInstruction>,
-    pub receive_pty_instructions: Receiver<(PtyInstruction, ErrorContext)>,
+pub struct Pty {
+    pub bus: Bus<PtyInstruction>,
     pub id_to_child_pid: HashMap<RawFd, RawFd>,
-    os_input: Box<dyn OsApi>,
     debug_to_file: bool,
     task_handles: HashMap<RawFd, JoinHandle<()>>,
 }
 
+fn handle_event(
```

Comment:
We could potentially merge this into `thread_main` for consistency with the other threads :)

Reply:
So the lack of consistency is because the first thing the pty thread does when we kick it off is to send itself a `PtyInstruction::NewTab`. I'd rather push for consistency the other way - i.e. add `handle_event` functions for the other thread main functions. Do you think that would be an imprveoemtn?

Reply:
Yeah, I do think that consistency is good here, so I'm fine with pushing that consistency the other way too :)

Reply:
Kunal's changes for server stuff means that the pty thread doesn't need to self-pulse any more, so I've gone without the `handle_event` function. :)